File: legato-java/app/src/main/java/mx/yellowme/java/legato/handlers/SessionFlowHandler.java
package mx.yellowme.java.legato.handlers;

import android.support.annotation.NonNull;

import mx.yellowme.java.legato.presentation.dispatch.DispatchContract;
import mx.yellowme.java.legato.utils.DataFaker;


public class SessionFlowHandler extends FlowHandler<DataFaker, DispatchContract.Screen> {

    public SessionFlowHandler(@NonNull DataFaker dataFaker, @NonNull DispatchContract.Screen screen) {
        super(dataFaker, screen);
    }

    @Override
    public void runValidations(FlowHandlerCallback callback) {
        if (logicHandler.hasSessionStarted()) {
            executeNext(callback);
        } else {
            executeEarlyExit(callback);
        }
    }
}

File: Legato/Legato/Handlers/SessionFlowHandler.swift
//
//  SessionFlowHandler.swift
//  Legato
//
//  Created by Luis Burgos on 7/23/18.
//  Copyright © 2018 Yellowme. All rights reserved.
//

import Foundation

class SessionFlowHandler: FlowHandler<DataProvider, Flows> {
    override func runValidations(_ callback: FlowHandlerCallback) {
        if (logicHandler as! SessionFaker).hasSessionStarted() {
            executeNext(callback)
        } else {
            executeEarlyExit(callback)
        }
    }
}
